### ISSUE TITLE
Create real-image preprocessing gallery

### ISSUE THREAD

**Issue by tomotvos** [tomotvos/ronchi-ml#14]
## Goal

Create a static visual gallery showing real-image preprocessing stages for inspection.

## Context

Real Ronchi images can be confounded by illumination gradients, contrast variation, thresholding artifacts, inversion, and cropping. A gallery makes it easier to inspect whether preprocessing preserves actual fringe geometry.

## Relevant files

- src/preprocess.py, src/preprocess_v2.py, src/preprocess_real.py
- src/topfix.py
- data/real/ — real image samples
- docs/preprocessing-pipeline.md

## Non-goals

- Do not build a web app.
- Do not infer model quality from the gallery alone.
- Do not alter training data.

## Proposed approach

For each real image, save a panel containing: original image, grayscale, normalized grayscale, illumination-corrected image if used, thresholded/binarized image, final model input.

Create an index file: runs/preprocess/<run-id>/gallery/index.md

## Acceptance criteria

- Gallery command accepts an input directory.
- Gallery command writes static images.
- Gallery command writes an index Markdown file.
- Gallery includes preprocessing parameters.
- Failed images are reported clearly without crashing the entire run.

## Validation command or artifact

```bash
python tools/make_preprocessing_gallery.py \
  --input-dir data/real \
  --output-dir runs/preprocess/gallery-test
```

## Notes for future coding agent

This should be simple and boring. Static PNGs plus Markdown are enough.
